### PR TITLE
✨ API Integration and New components

### LINKED ISSUE #6: Integrate backend from pathivu
- Add store support for listing microservices and integrate with backend
- GET `http:// localhost:{port}/query`  to visualise the response

### PR DESCRIPTION
**Pull request by rahulakrishna**
🎨 New components. 
   🎨 Date selection module
   🎨 Config
   🎨 Multi select Sidebar
✨ Search functionality

Resolves #6

### CODE REVIEW

File: src/stylesheets/components/datepicker.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
+@import '../variables.scss';
+
+.datepicker-container {
+  position: fixed;
+  top: 10px;
+  left: 10px;
+  right: 10px;
+  background: #222231;
```
Comment (rishiraj824) [inline src/stylesheets/components/datepicker.scss]: For most of the values in these files, we can declare and use variables from `variables.scss`

File: src/stylesheets/containers/config.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
+@import '../variables.scss';
+
+.config-container {
+  position: fixed;
+  top: 20px;
+  left: 20px;
+  right: 20px;
+  width: auto;
+  height: 90vh;
+  background: #222231;
```
Comment (rishiraj824) [inline src/stylesheets/containers/config.scss]: Same for these styles.

File: src/stylesheets/containers/config.scss [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  padding: 20px;
+  z-index: 1000;
+  .header {
+    display: flex;
```
Comment (rishiraj824) [inline src/stylesheets/containers/config.scss]: Probably we can come up with mixins
`@mixin center {
 justify-content:center;
align-items:center;
}`
`@mixin flex-row {
 display: flex;
flex-direction: row;
}`

File: src/stylesheets/containers/config.scss [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+.config-container {
+  position: fixed;
+  top: 20px;
+  left: 20px;
+  right: 20px;
+  width: auto;
+  height: 90vh;
+  background: #222231;
+  padding: 20px;
+  z-index: 1000;
```
Comment (rishiraj824) [inline src/stylesheets/containers/config.scss]: Let's keep this in order. It escalates to higher values really quickly.

File: src/stylesheets/containers/config.scss [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      margin-top: 10px;
+    }
+  }
+  .footer {
+    position: absolute;
+    bottom: 30px;
+    width: 95%;
+    display: flex;
+    flex-direction: row-reverse;
+    .submit-button {
```
Comment (rishiraj824) [inline src/stylesheets/containers/config.scss]: We can reuse this class I think let's keep it in a `commons.scss` file or `buttons.scss`

File: src/utils/time.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import moment from 'moment';
+
+let now = Date.now();
+
+export const TIME_MAPPING = {
```
Comment (rishiraj824) [inline src/utils/time.js]: 💯

File: src/stylesheets/containers/visualiser.scss [hunk 7]
```diff
@@ -1,8 +1,59 @@
...
+    pre {
+      font-family: Fira Code;
+      font-size: 14px;
+      max-width: 600px;
+      overflow: scroll;
+      white-space: pre;
+      overflow-wrap: break-word;
+    }
+  }
+  .date-time {
```
Comment (rishiraj824) [inline src/stylesheets/containers/visualiser.scss]: We will use dates a lot makes sense to keep somewhere common or in a separate `date-picker.scss` file. What do you think?

File: src/stylesheets/containers/config.scss [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      transition: all 0.5s;
+      cursor: pointer;
+      padding-right: 16px;
+      padding-top: 8px;
+      padding-bottom: 8px;
+      border-radius: 8px;
+      font-size: 22px;
+      background: #092d09;
+      color: white;
+      font-family: Ubuntu;
```
Comment (rishiraj824) [inline src/stylesheets/containers/config.scss]: font-family: $font-family.

File: src/stylesheets/components/select.scss [hunk 9]
```diff
@@ -1,8 +1,46 @@
...
+  -webkit-appearance: none;
+  background: #00a8b0;
+  width: 85%;
+  color: #fff;
+  padding: 8px 20px;
+  outline: 0;
+}
+
+.dropdown {
+  width: 180px;
```
Comment (rishiraj824) [inline src/stylesheets/components/select.scss]: 💯

File: src/stylesheets/components/searchbox.scss [hunk 10]
```diff
@@ -1,19 +1,60 @@
...
+  align-items: flex-start;
+  .input {
     -webkit-appearance: none;
     box-sizing: border-box;
     border: 0;
     background: $input-background;
     padding: 1.2rem 1rem;
     border-radius: 8px;
-    width: 100%;
+    width: 80%;
```
Comment (rishiraj824) [inline src/stylesheets/components/searchbox.scss]: Wonder whether these random widths will cause problems.

File: src/stylesheets/components/searchbox.scss [hunk 11]
```diff
@@ -1,19 +1,60 @@
 @import '../variables.scss';
 
-.input {
+.searchbar-container {
+  display: flex;
+  flex-direction: row;
```
Comment (rishiraj824) [inline src/stylesheets/components/searchbox.scss]: not needed I guess

File: src/containers/Sidebar.js [hunk 12]
```diff
@@ -1,24 +1,70 @@
...
 
+const Sidebar = ({
+  partitions,
+  fetchPartitions,
+  setInput,
+  selectedPartitions,
+  hostUrl
+}) => {
+  useEffect(() => {
+    console.log({ componentSays: hostUrl });
```
[the remark this answers is not in the document]
Comment (rahulakrishna) [reply src/containers/Sidebar.js]: Resolved

File: src/stylesheets/containers/visualiser.scss (hunk 7 above)
Comment (rahulakrishna) [reply src/stylesheets/containers/visualiser.scss]: We have a date-picker component. This is specific to the date option inside search area

**Pull request merged**